### ISSUE TITLE
Crash while spawning villager mobs [2.8 only; Fixed in v3.2.2]

### ISSUE THREAD

**Issue by Thijmer** [TheDuckCow/MCprep#74]
***Update from TheDuckCow***
Confirmed bug where spawning any of the Mobs listed below will cause an immediate crash and close of Blender 2.8, currently known to only affect Blender 2.8 builds since sometime after May 7th 2019 (known for sure to affect 6/13 and 7/16 builds, and the first 2.8 release candidate).

Only affected mobs (all others tested)
- Villager
- Vindicator
- Evoker

For this reason, **be cautious when mob spawning** - press save on your file before running, especially if you are also using custom installed mobs. The issue is tied to the raw asset file, so it is not on it's own a code-update issue, the assets must be updated once a fix is determined.

Alternatively, consider using a "custom mobs" and install the [mob rigs from BoxScape studios](https://sites.google.com/view/boxscape-studios/downloads?authuser=0) - I have tested these import fine for Blender 2.8 (though materials will not look correct in 2.7 Blender Internal).

***Original message below***

Hi, Here's a issue: 
Almost always when I spawn a mob, blender crashes immediately.
I have a **zipped MP4** attatched, so that you can see it happen.
The strange thing is that it doesn't always happen.

Perhaps this helps you solving the issue:
-I'm using the **Blender 2.80** release from 2019-07-05.
-It happens more often when there is already a MC world loaded
-Intel I9, 32 GB ram, Windows 10
-I've tried it with a villager, a zombie, a witch, a wolf, a creeper
-Only the wolf succeeded, but despite the success, there was still an error. Luckly no crash.
-It also happens if I have the new 3D cursor rotation optin set to x=0, y=0, z=0 rotation

I hope you can recreate and fix the error, it's really annoying.

[crashreport.zip](https://github.com/TheDuckCow/MCprep/files/3389693/Video.for.crashreport.zip)

I hope my English was not too bad, I'm still learning :)

**Comment by TheDuckCow** [comment]
Thanks for reporting the issue! That’s a really bad thing and I definitely want to resolve this asap. Could you try a couple things for me?

1) Worth a try, completely remove (ie press remove, not just disable the checkbox in preferences) the addon, close blender, and then install the latest again <https://theduckcow.com/dev/blender/mcprep-download/>.
2) Try using the latest Blender 2.8 build (not the release candidate, but the daily build here <https://builder.blender.org/download/>). If it works then please let me know.
3) Can you screen record just like you did, but have the python console open (From: Info bar, Window, Show Console)? And have the main blender window shifted to the side some so that the console window is revealed. Normally I’d ask just for a screenshot of the console output, but if blender crashing you’ll have a hard time doing this, hence the screen recording will work best.
4) Can you see if it works fine on your machine using blender 2.79 for instance? If it also doesn’t work there, it implies something different and will help me debug what your issue might be.

Thanks for bearing with me,
Patrick

**Comment by Thijmer** [comment]
Dear Mr Patrick,
I update my blender 2.80 builds dayly, and I’m now running the release candidate. Some mobs still won’t spawn, like the villager, but others work fine (e.g. skeleton, player).

Here are the results of your suggestions:
1) I removed the addon and reinstalled it from ZIP, tried to spawn a Vindicator, Blender crashed immediately.
2) I installed the latest beta version of Blender, tried to spawn a Vindicator again, immediately a crash. I noticed that a Vindicator or a villager didn’t work, but a tropical fish worked fine. Should I try this with more mobs?
NOTICE: From now on I’m using the latest BETA build an not the release candidate.
3) Here you go! (This time I uploaded it to YouTube, the limit of the attachment size is so annoying! Don't worry, it's hidden, only people with the link can access it.)
4) I downloaded Blender 2.79, installed the addon from ZIP and it actually worked! I tried the villager, the vindicator and some other mobs and they all worked! I think the error is at the 2.80 compatibility. You too?

I’m glad I can help you debugging this Blender plugin. It’s an amazing plugin!
And I hope you can find and fix the error soon.

Kind regards,
Thijmen Voskuilen (a Dutch name, because I live in the Netherlands)


Van: Patrick W. Crawford
Verzonden: maandag 15 juli 2019 10:33
Aan: TheDuckCow/MCprep
CC: Thijmer; Author
Onderwerp: Re: [TheDuckCow/MCprep] Crash while spawning mobs (#74)

Thanks for reporting the issue! That’s a really bad thing and I definitely want to resolve this asap. Could you try a couple things for me?

1) Worth a try, completely remove (ie press remove, not just disable the checkbox in preferences) the addon, close blender, and then install the latest again <https://theduckcow.com/dev/blender/mcprep-download/>.
2) Try using the latest Blender 2.8 build (not the release candidate, but the daily build here <https://builder.blender.org/download/>). If it works then please let me know.
3) Can you screen record just like you did, but have the python console open (From: Info bar, Window, Show Console)? And have the main blender window shifted to the side some so that the console window is revealed. Normally I’d ask just for a screenshot of the console output, but if blender crashing you’ll have a hard time doing this, hence the screen recording will work best.
4) Can you see if it works fine on your machine using blender 2.79 for instance? If it also doesn’t work there, it implies something different and will help me debug what your issue might be.

Thanks for bearing with me,
Patrick

**Comment by TheDuckCow** [comment]
Hi @Thijmer,

Thanks for working through all these steps. I didn't get the video link in your last message, but I might not need it now afterall:

The "good news" is that I'm also now seeing the issue as well. I am able to use older versions of blender 2.8 to spawn e.g. the villager, but there seems to be issues with the raw rig files themselves (these blend files in fact crash on open, independent to MCprep) and opening in more recent versions of Blender 2.8

So for now, try using an older build of 2.8 to import the model, save the file, and then continue on in a _later_ version of 2.8 (I confirmed that appears to work).

*Following are notes to myself to help resolve the issue*

- Find out all of the rigs which are affected (any help here is appreciated!) Known so far (ie **spawning these in Blender 2.8 latest builds will immediately crash Blender**):
  - Villager
  - Vindicator
  - Evoker
- Not broken (ie crash inducing) but noting to self it spawns weird right now: Blaze
- (interesting note: the Witch is *not* affected)
- Note log outputs in console below when opening these rigs in older versions of Blender 2.8 (ie where it doesn't crash immediately), I have a feeling this is the root cause
- Also worth noting: the only files that appear affected right now in reality were all created from the same original BoxScape Villager file (I just did some clean up/splitting up into separate blends).
- Worth calling out that many other mobs have other console warnings too, e.g. dependency cycles. Lower priority and not blocking right now.

```
ERROR (bke.anim_sys): /Users/gast/blender-buildbot/mac_x86_64_10_9_cmake/blender.git/source/blender/blenkernel/intern/anim_sys.c:4133 BKE_animsys_eval_driver: invalid driver - hide[0]
ERROR (bke.anim_sys): /Users/gast/blender-buildbot/mac_x86_64_10_9_cmake/blender.git/source/blender/blenkernel/intern/anim_sys.c:4133 BKE_animsys_eval_driver: invalid driver - hide[0]
ERROR (bke.anim_sys): /Users/gast/blender-buildbot/mac_x86_64_10_9_cmake/blender.git/source/blender/blenkernel/intern/anim_sys.c:4133 BKE_animsys_eval_driver: invalid driver - pose.bones["Bone.016"].constraints["IK"].influence[0]
ERROR (bke.object): /Users/gast/blender-buildbot/mac_x86_64_10_9_cmake/blender.git/source/blender/blenkernel/intern/object.c:2306 ob_parbone: Object Armature.033 with Bone parent: bone Bone.017 doesn't exist
ERROR (bke.anim_sys): /Users/gast/blender-buildbot/mac_x86_64_10_9_cmake/blender.git/source/blender/blenkernel/intern/anim_sys.c:4133 BKE_animsys_eval_driver: invalid driver - pose.bones["Bone.013"].constraints["IK"].influence[0]
ERROR (bke.anim_sys): /Users/gast/blender-buildbot/mac_x86_64_10_9_cmake/blender.git/source/blender/blenkernel/intern/anim_sys.c:4133 BKE_animsys_eval_driver: invalid driver - key_blocks["Key 4"].value[0]
ERROR (bke.anim_sys): /Users/gast/blender-buildbot/mac_x86_64_10_9_cmake/blender.git/source/blender/blenkernel/intern/anim_sys.c:4133 BKE_animsys_eval_driver: invalid driver - key_blocks["Key 6"].value[0]
ERROR (bke.anim_sys): /Users/gast/blender-buildbot/mac_x86_64_10_9_cmake/blender.git/source/blender/blenkernel/intern/anim_sys.c:4133 BKE_animsys_eval_driver: invalid driver - key_blocks["Key 6"].value[0]
ERROR (bke.anim_sys): /Users/gast/blender-buildbot/mac_x86_64_10_9_cmake/blender.git/source/blender/blenkernel/intern/anim_sys.c:4133 BKE_animsys_eval_driver: invalid driver - key_blocks["Key 6"].value[0]
ERROR (bke.anim_sys): /Users/gast/blender-buildbot/mac_x86_64_10_9_cmake/blender.git/source/blender/blenkernel/intern/anim_sys.c:4133 BKE_animsys_eval_driver: invalid driver - key_blocks["Key 6"].value[0]
ERROR (bke.modifier): /Users/gast/blender-buildbot/mac_x86_64_10_9_cmake/blender.git/source/blender/blenkernel/intern/modifier.c:410 modifier_setError: From/To objects must be set
```

**Comment by TheDuckCow** [comment]
Minor update. I've been looking at the villager rig, and I found that if I delete all mesh objects from the file, then it won't crash - but I have not been able to get it to not crash (in 2.8) by only deleting some objects (my hope was that the issue was contained to a single object datablock that once removed would resolve the issue, but alas).

Will continue to investigate, but if the issue cannot be resolved I may consider rolling  back to the previous villager rig I myself created, and spend the tie myself to create from this the Vindicator and Evoker rigs to avoid this issue altogether. If I do this, it will be part of a new release entirely; if I'm able to update the existing assets fixing the issue, then I'll update the existing release 3.2.0 resources.

**Comment by TheDuckCow** [comment]
This has been fixed as of the 3.2.2 release now out. Will keep this bug open for a little bit for visibility and to encourage people to update.

**Comment by TheDuckCow** [comment]
Marking this resolved as this has long been fixed.
